openlibrary/plugins/openlibrary/code.py

```diff
@@ -402,15 +402,9 @@ class robotstxt(delegate.page):
 
     def GET(self):
         web.header('Content-Type', 'text/plain')
-        try:
-            is_dev = (
-                'dev' in infogami.config.features or web.ctx.host != 'openlibrary.org'
-            )
-            robots_file = 'norobots.txt' if is_dev else 'robots.txt'
-            data = open('static/' + robots_file).read()
-            raise web.HTTPError('200 OK', {}, data)
-        except OSError:
-            raise web.notfound()
+        is_dev = 'dev' in infogami.config.features or web.ctx.host != 'openlibrary.org'
+        robots_file = 'norobots.txt' if is_dev else 'robots.txt'
+        return web.ok(open(f'static/{robots_file}').read())
```

Comment:
If `pathlib.Path().read_text()` is not used then the Python language spec does not guarantee that unclosed file handles will ever be cleaned up.
```suggestion
        with open(f'static/{robots_file}') as in_file:
            return web.ok(in_file.read())
```
Repeat below...

Reply:
Oh I love Path 😋 That's kind of unrelated to this change though ; we should do that in a separate PR